library/src/main/java/com/bumptech/glide/load/resource/bitmap/ImageReader.java

```diff
@@ -43,17 +43,25 @@ final class ByteArrayReader implements ImageReader {
     private final byte[] bytes;
     private final List<ImageHeaderParser> parsers;
     private final ArrayPool byteArrayPool;
+    private final boolean enableHardwareGainmapFixOnU;
 
-    ByteArrayReader(byte[] bytes, List<ImageHeaderParser> parsers, ArrayPool byteArrayPool) {
+    ByteArrayReader(
+        byte[] bytes,
+        List<ImageHeaderParser> parsers,
+        ArrayPool byteArrayPool,
+        boolean enableHardwareGainmapFixOnU) {
       this.bytes = bytes;
       this.parsers = parsers;
       this.byteArrayPool = byteArrayPool;
+      this.enableHardwareGainmapFixOnU = enableHardwareGainmapFixOnU;
     }
 
     @Nullable
     @Override
     public Bitmap decodeBitmap(Options options) {
-      return BitmapFactory.decodeByteArray(bytes, /* offset= */ 0, bytes.length, options);
+      return enableHardwareGainmapFixOnU
```

Comment:
The goal here and below is that once enableHardwareGainmapFixOnU is permanently shipped, we remove enableHardwareGainmapFixOnU and only call into GlideBitmapFactory, right?

Reply:
Yes, exactly. We will clean it up as we would any Glide flag.